### PR TITLE
maintenance(www) upgrade jest and testing-library

### PR DESCRIPTION
**Pull request by tesseralis**
## Description

Upgrade jest and react testing library to latest versions, and fix some setup files.

### CODE REVIEW

File: www/package.json [hunk 1]
```diff
@@ -162,8 +162,7 @@
     "front-matter": "^2.3.0",
     "github-api": "^3.3.0",
     "identity-obj-proxy": "^3.0.0",
-    "jest": "^24.9.0",
-    "jest-dom": "^3.5.0",
```
Comment (tesseralis) [inline www/package.json:166]: `jest-dom` is deprecated in favor of `@testing-library/jest-dom`.

File: www/jest.config.js [hunk 2]
```diff
@@ -6,15 +6,12 @@ module.exports = {
...
-  testPathIgnorePatterns: [`node_modules`, `.cache`],
+  testPathIgnorePatterns: [`node_modules`, `.cache`, `public`],
   transformIgnorePatterns: [`node_modules/(?!(gatsby)/)`],
   globals: {
     __PATH_PREFIX__: ``,
   },
   testURL: `http://localhost`,
   setupFiles: [`<rootDir>/loadershim.js`],
-  setupFilesAfterEnv: [
-    `@testing-library/react/cleanup-after-each`,
```
Comment (tesseralis) [inline www/jest.config.js:17]: This is no longer needed; running it gives an deprecation warning in the console.

File: www/jest.config.js [hunk 3]
```diff
@@ -6,15 +6,12 @@ module.exports = {
...
-    `jest-dom/extend-expect`,
```
Comment (tesseralis) [inline www/jest.config.js:18]: This and just importing `jest-dom` are [exactly the same](https://github.com/testing-library/jest-dom/blob/master/src/index.js)

File: www/jest.config.js [hunk 4]
```diff
@@ -6,15 +6,12 @@ module.exports = {
     ".+\\.(css|styl|less|sass|scss)$": `identity-obj-proxy`,
     ".+\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": `<rootDir>/__mocks__/file-mock.js`,
   },
-  testPathIgnorePatterns: [`node_modules`, `.cache`],
+  testPathIgnorePatterns: [`node_modules`, `.cache`, `public`],
```
Comment (tesseralis) [inline www/jest.config.js:9]: Not adding `public` was causing jest to error out when I tried to run the tests when there was an existing build.

File: www/package.json (hunk 1 above)
Comment (tesseralis) [reply www/package.json:166]:
```
warning jest-dom@3.5.0: 🚨 jest-dom has moved to @testing-library/jest-dom. Please uninstall jest-dom and install @testing-library/jest-dom instead, or use an older version of jest-dom. Learn more about this change here: https://github.com/testing-library/dom-testing-library/issues/260 Thanks! :)
```

File: www/jest.config.js (hunk 3 above)
Comment (DSchau) [reply www/jest.config.js:18]: Today I learned!

**Pull request merged**